core/include/byteorder.h

```diff
@@ -445,22 +445,13 @@ static inline uint64_t ntohll(uint64_t v)
 
 static inline uint16_t byteorder_bebuftohs(const uint8_t *buf)
 {
-#if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
-    return (uint16_t)((buf[0] << 8) | buf[1]);
-#else
-    return (uint16_t)((buf[1] << 8) | buf[0]);
```

Comment:
I agree the old version was wrong, this line is reading a 16 bit integer from two bytes in little endian byte ordering